### ISSUE TITLE
synchronous scraping of a dynamic (javascript) rendered page

### ISSUE THREAD

**Issue by Offbeatmammal** [hardkoded/puppeteer-sharp#1769]
trying to run Puppeteer synchronously as I need the results of the 'scraped' page so await is a pain (and happy to wait on execution).

_**If I use the code below with headless = false, it works and I get my required value.
If I use headless = true it never completes.**_ Would _really_ prefer to run headless!

Is there a better way to do this (had a look through docs, but not luck)? The 'trigger' for a page completing is the addition (as the final step of the javascript code on that page) of a `<div id='done'></div>` if that would help (though the WaitForSelector doesn't seem to help)

```

Task[] Tasks = new Task[1];
            Tasks[0] = Task.Run(async () =>
            {
                await new BrowserFetcher().DownloadAsync(BrowserFetcher.DefaultChromiumRevision);
                browser = await Puppeteer.LaunchAsync(new LaunchOptions
                {
                    Headless = true
                });
                var page = await browser.NewPageAsync();
                var navigation = new NavigationOptions
                {
                    Timeout = 0,
                    WaitUntil = new[] {
                        WaitUntilNavigation.DOMContentLoaded }
                };
                await page.GoToAsync(fullUrl, navigation);
                //await page.WaitForSelectorAsync("done");
                var a3 = await page.EvaluateFunctionAsync<string>($"()=>document.getElementById('output').innerHTML");

                val = a3;
                
            });
            
        Task.WaitAll(Tasks);
```

**Comment by kblok** [comment]
Can you try the [EnqueueAsyncMessages](https://www.puppeteersharp.com/api/PuppeteerSharp.LaunchOptions.html#PuppeteerSharp_LaunchOptions_EnqueueAsyncMessages) option?
